pkgs/tools/security/gnupg/22.nix

```diff
@@ -50,7 +50,14 @@ stdenv.mkDerivation rec {
     "--with-npth-prefix=${npth}"
   ] ++ optional guiSupport "--with-pinentry-pgm=${pinentry}/${pinentryBinaryPath}";
 
-  postInstall = ''
+  postInstall = if enableMinimal
```

Comment:
please use `stdenv.lib.optionalString` here.

We also need to keep the symlink to `gpg2` around? We point to it inside systemd.

Reply:
Can you give me an example for that? I thought that it would append string b to string a if condition is true. I don't want that, as there is no point in copying files which shall be delete then again.

Regarding the `gpg2` binary: I don't think that's necessary
https://github.com/systemd/systemd/blob/2d901d33a90ef9d3fe01ac66c4894c9e6bf48ce0/src/import/pull-common.c#L493 calls `gpg2` first, then `gpg`. Because of that, having `gpg` only is fine, though having `gpg2` only probably fails.

Reply:
example for `optionalString` usage: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/system/netdata/default.nix#L43

Reply:
Ah yes. So that is not what we need there?!

Reply:
In the end we do something similar, but `optionalString` is more commonly used and readable IMHO.

Reply:
I'm afraid you didn't get my point (there is no point in copying files which shall be delete then again => it is  xor for the two `postInstall` strings), or I didn't understand `optionalString`. Please show me how you would change the code, then we can know for sure what we are talking about.

Reply:
You're right - `optionalString` probably doesn't make a lot of sense here. My bad.